Model/CustomerInformation.php

```diff
@@ -0,0 +1,123 @@
+        Request                         $request,
+        CustomerRepositoryInterface     $customerRepository,
+        Logger                          $customerLog,
+        CollectionFactory               $orderCollectionFactory,
+        Customer                        $customerCollection,
+        PaymentTokenManagementInterface $paymenttokenmanagement
+    ) {
+        $this->request                = $request;
+        $this->customerRepository     = $customerRepository;
+        $this->customerLog            = $customerLog;
+        $this->orderCollectionFactory = $orderCollectionFactory;
+        $this->customerCollection     = $customerCollection;
+        $this->paymenttokenmanagement = $paymenttokenmanagement;
+    }
+
+    public function getCustomerInformation()
+    {
+        try {
+            $body       = $this->request->getBodyParams();
+            $customerId = $body['data']['customer_id'];
+
+            // Retrieve customer information by ID
+            $customer      = $this->customerRepository->getById($customerId);
+            $id            = $customer->getId();
+            $email         = $customer->getEmail();
+            $createdAt     = $customer->getCreatedAt();
+            $createdDate   = date('m/d/Y', strtotime($createdAt));
+            $lastLoginAt   = $this->customerLog->get($customerId);
+            $lastLoginDate = date('m/d/Y', strtotime($lastLoginAt->getLastLoginAt()));
+
+            // Retrieve customer order collection
+            $orderCollection = $this->orderCollectionFactory->create()
+                ->addFieldToFilter('customer_id', $customerId)
+                ->addAttributeToSort('entity_id', 'DESC');
+            $remoteIp               = "";
+            $lastOrderDateFormatted = "";
+            $totalOrders            = "";
+            $totalOrderAmount       = "";
+            if (count($orderCollection)) {
+                // Retrieve details from the last order
+                $lastOrder     = $orderCollection->getFirstItem();
+                $remoteIp      = $lastOrder->getRemoteIp();
+                $lastOrderDate = $lastOrder->getCreatedAt();
+                $lastOrderDateFormatted = date('m/d/Y', strtotime($lastOrderDate));
+                $totalOrders       = $orderCollection->getSize();
+                $totalOrderAmounts = $orderCollection->getColumnValues('grand_total');
+                $totalOrderAmount  = array_sum($totalOrderAmounts);
+            }
+
+            // Retrieve additional customer information
+            $customersCollection = $this->customerCollection->getCollection()
+                ->addAttributeToSelect("*")
+                ->addAttributeToFilter("entity_id", $customerId)
+                ->load();
+            $customerInformationAll = $customersCollection->getData();
+            $customerInformation = array();
+            foreach ($customerInformationAll as $customerInfo) {
+                unset($customerInfo['password_hash']);

```

Comment:
Let's also remove `rp_token`, we don't need it so it's best not to expose it.

Reply:
Hi @soleilcot , I've added the recommenedation. see here https://github.com/Nofraud/nofraud_magento_checkout/pull/41/files